tests/measurements/test_classical_shadow.py

```diff
@@ -223,6 +224,197 @@ def test_expval_same_rng(self):
+        assert set(res[1].ravel()).issubset({0, 1, 2})
+
+    def test_wire_order(self, seed):
+        """Test that the wire order is respected"""
+        state = np.array([[1, 1], [0, 0]]) / np.sqrt(2)
+        dm = np.outer(state, state).reshape((2,) * 4)
+
+        mp = qml.classical_shadow(wires=[0, 1])
+        res = mp.process_density_matrix_with_shots(
+            dm, qml.wires.Wires([0, 1]), shots=1000, rng=seed
+        )
+
+        assert res.shape == (2, 1000, 2)
+        assert res.dtype == np.int8
+
+        # test that the first qubit samples are all 0s when the recipe is Z
+        assert np.all(res[0][res[1, ..., 0] == 2][:, 0] == 0)
+
+        # test that the second qubit samples contain 1s when the recipe is Z
+        assert np.any(res[0][res[1, ..., 1] == 2][:, 1] == 1)
+
+        res = mp.process_density_matrix_with_shots(
+            dm, qml.wires.Wires([1, 0]), shots=1000, rng=seed
+        )
+
+        assert res.shape == (2, 1000, 2)
+        assert res.dtype == np.int8
+
+        # now test that the first qubit samples contain 1s when the recipe is Z
+        assert np.any(res[0][res[1, ..., 0] == 2][:, 0] == 1)
+
+        # now test that the second qubit samples are all 0s when the recipe is Z
+        assert np.all(res[0][res[1, ..., 1] == 2][:, 1] == 0)
+
+    @pytest.mark.parametrize("num_wires", [3, 4, 5])
+    @pytest.mark.parametrize("shots", [100])
+    def test_wire_order_generalized(self, num_wires, shots, seed):
+        """Test that the wire order is respected for any number of wires and all permutations"""
+        # Prepare a GHZ-like state for num_wires
+        state = np.zeros((2,) * num_wires)
+        idx0 = tuple([0] * num_wires)
+        idx1 = tuple([1] * num_wires)
+        state[idx0] = 1 / np.sqrt(2)
+        state[idx1] = 1 / np.sqrt(2)
+        dm = np.outer(state.ravel(), state.ravel()).reshape((2,) * (2 * num_wires))
+
+        wires = list(range(num_wires))
+        mp = qml.classical_shadow(wires=wires)
+
+        for perm in itertools.permutations(wires):
+            res = mp.process_density_matrix_with_shots(
+                dm, qml.wires.Wires(list(perm)), shots=shots, rng=seed
+            )
+            assert res.shape == (2, shots, num_wires)
+            # bits are always 0 or 1
+            assert set(res[0].ravel()).issubset({0, 1})
+            # recipes are always 0, 1, or 2
+            assert set(res[1].ravel()).issubset({0, 1, 2})
+
+            # For GHZ state, when all qubits are measured in Z basis,
+            # they should all have the same value (all 0s or all 1s)
+            z_shots = np.where(np.all(res[1] == 2, axis=1))[0]
+            if len(z_shots) > 0:  # Only test if we have any all-Z shots
+                for shot_idx in z_shots:
+                    # All qubits should have the same measurement outcome
+                    first_bit = res[0][shot_idx, 0]
+                    assert np.all(res[0][shot_idx] == first_bit)
```

Comment:
@dwierichs I added a test here to make use of GHZ s.t. we can have deterministic results. However as I increased the `num_wires` the time cost still increased fast so stopped at `5`.

Reply:
Oh wow, yeah that's quite a general test :D I'm not surprised that this blows up very quickly: The density matrix scales as $4^n$, the number of contractions scales with $n$ and the number of permutations is $n!$, so that's a scaling of at least $n 4^n (n!)$, which is about as horrible as something can scale in real-world code, I suppose :D